### ISSUE TITLE
[Bug]: [benchmark]diskann indexing, after inserting 10 million data, load collection, querynode restarted 8 times

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#26184]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230807-4e1b65d3
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task : fouramf-stable-test-5jr87, id : 1
case: test_concurrent_locust_diskann_dql_cluster

This case is to insert 10 million data, load collection, and then concurrently query, load, search; but observed that querynode restarted 8 times during load collection. 

server:
Problem: querynode rebooted 8 times
```
fouramf-stable-test-5jr87-1-etcd-0                                1/1     Running     0               127m    10.104.15.35    4am-node20   <none>           <none>
fouramf-stable-test-5jr87-1-etcd-1                                1/1     Running     0               127m    10.104.24.245   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-etcd-2                                1/1     Running     0               127m    10.104.14.77    4am-node18   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-datacoord-78d6bbf77c-5gdww     1/1     Running     0               127m    10.104.9.167    4am-node14   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-datanode-5854688bb5-zfkf5      1/1     Running     1 (123m ago)    127m    10.104.5.230    4am-node12   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-indexcoord-b5f4d6cc6-z9czr     1/1     Running     0               127m    10.104.24.235   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-indexnode-7779f99f55-zj8zq     1/1     Running     0               127m    10.104.15.31    4am-node20   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-proxy-59bfb4b596-mjpgh         1/1     Running     1 (123m ago)    127m    10.104.24.234   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-querycoord-57b6d5f6b7-qdn8m    1/1     Running     1 (123m ago)    127m    10.104.24.236   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-querynode-7bb98b5877-8qsmw     1/1     Running     8 (67m ago)     127m    10.104.9.168    4am-node14   <none>           <none>
fouramf-stable-test-5jr87-1-milvus-rootcoord-7cd65b867f-sfplv     1/1     Running     1 (123m ago)    127m    10.104.18.201   4am-node25   <none>           <none>
fouramf-stable-test-5jr87-1-minio-0                               1/1     Running     0               127m    10.104.15.33    4am-node20   <none>           <none>
fouramf-stable-test-5jr87-1-minio-1                               1/1     Running     0               127m    10.104.19.24    4am-node28   <none>           <none>
fouramf-stable-test-5jr87-1-minio-2                               1/1     Running     0               127m    10.104.24.246   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-minio-3                               1/1     Running     0               127m    10.104.14.78    4am-node18   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-bookie-0                       1/1     Running     0               127m    10.104.5.243    4am-node12   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-bookie-1                       1/1     Running     0               127m    10.104.18.214   4am-node25   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-bookie-2                       1/1     Running     0               127m    10.104.17.223   4am-node23   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-bookie-init-4w84h              0/1     Completed   0               127m    10.104.15.30    4am-node20   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-broker-0                       1/1     Running     0               127m    10.104.24.237   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-proxy-0                        1/1     Running     0               127m    10.104.18.203   4am-node25   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-pulsar-init-5ww9v              0/1     Completed   0               127m    10.104.18.204   4am-node25   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-recovery-0                     1/1     Running     0               127m    10.104.18.202   4am-node25   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-zookeeper-0                    1/1     Running     0               127m    10.104.14.73    4am-node18   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-zookeeper-1                    1/1     Running     0               125m    10.104.24.250   4am-node29   <none>           <none>
fouramf-stable-test-5jr87-1-pulsar-zookeeper-2                    1/1     Running     0               124m    10.104.19.26    4am-node28   <none>           <none>
```

querynode memory usage :   
Problem: As you can see, querynode's memory monitoring is not continuous and is restarted 8 times during the load phase (5:00 to 5:19) 
![image](https://github.com/milvus-io/milvus/assets/48523564/c9b1f5fc-1b0e-441c-8a75-449944a40ec9)

client log (In conjunction with grafana's monitoring, the load collection is running from 5:00 to 5:20. ):
```
[2023-08-07 04:28:34,684 -  INFO - fouram]: [Base] Start build index of DISKANN for collection fouram_lr25Jv1M, params:{'index_type': 'DISKANN', 'metric_type': 'L2', 'params': {}} (base.py:442)
[2023-08-07 04:59:43,609 -  INFO - fouram]: [Time] Index run in 1868.9221s (api_request.py:45)
[2023-08-07 04:59:43,610 -  INFO - fouram]: [CommonCases] RT of build index DISKANN: 1868.9221s (common_cases.py:96)
[2023-08-07 04:59:43,613 -  INFO - fouram]: [Base] Params of index: [{'float_vector': {'index_type': 'DISKANN', 'metric_type': 'L2', 'params': {}}}] (base.py:456)
[2023-08-07 04:59:43,613 -  INFO - fouram]: [CommonCases] Prepare index DISKANN done. (common_cases.py:99)
[2023-08-07 04:59:43,613 -  INFO - fouram]: [CommonCases] No scalars need to be indexed. (common_cases.py:107)
[2023-08-07 04:59:43,615 -  INFO - fouram]: [Base] Number of vectors in the collection(fouram_lr25Jv1M): 10000000 (base.py:483)
[2023-08-07 04:59:43,615 -  INFO - fouram]: [Base] Start load collection fouram_lr25Jv1M,replica_number:1,kwargs:{} (base.py:298)
[2023-08-07 05:20:41,961 -  INFO - fouram]: [Time] Collection.load run in 1258.3453s (api_request.py:45)
[2023-08-07 05:20:41,967 -  INFO - fouram]: [Base] Describe resource group:__default_resource_group, ResourceGroupInfo:<name:__default_resource_group>,<capacity:1000000>,<num_available_node:1>,<num_loaded_replica:{'fouram_lr25Jv1M':1}>,<num_outgoing_node:{}>,<num_incoming_node:{}> (base.py:657)
```

### Expected Behavior

I don't think querynode should be rebooted. 

### Steps To Reproduce

```markdown
1. create a collection or use an existing collection
        2. build index on vector column  
        3. insert a certain number of vectors => 10m
        4. flush collection
        5. build index on vector column with the same parameters => diskann
        6. count the total number of rows
        7. load collection
        8. perform concurrent operations => (load、query、search)
        9. clean all collections
```

#### How to reproduce: 

Go to the "qtp" platform and run case : 'test_concurrent_locust_diskann_dql_cluster'

#### resources:
queryNode.resources.limits.cpu=4.0,queryNode.resources.limits.memory=6Gi,
queryNode.resources.requests.cpu=3.0,queryNode.resources.requests.memory=4Gi,queryNode.replicas=1,
indexNode.resources.limits.cpu=8.0,indexNode.resources.limits.memory=10Gi,
indexNode.resources.requests.cpu=5.0,indexNode.resources.requests.memory=6Gi,indexNode.replicas=1,
dataNode.resources.limits.cpu=2.0,dataNode.resources.limits.memory=4Gi,
dataNode.resources.requests.cpu=2.0,dataNode.resources.requests.memory=3Gi

**Comment by yanliang567** [comment]
/assign @jiaoew1991 
it seems that there are a few loading failures occurs in the last 2 days.

/unassign

**Comment by smellthemoon** [comment]
<img width="1427" alt="1" src="https://github.com/milvus-io/milvus/assets/64083300/7ff41377-43f5-4f83-89eb-eb8874981855">
querynode oom.

**Comment by smellthemoon** [comment]
pr merged, plz verify. @elstic

**Comment by elstic** [comment]
> pr merged, plz verify. @elstic

@smellthemoon  
This issue still exists, verify the image: master-20230809-d267559b

server:
```
fouramf-stable-test-1691618400-1-etcd-0                           1/1     Running     0                141m    10.104.4.155    4am-node11   <none>           <none>
fouramf-stable-test-1691618400-1-etcd-1                           1/1     Running     0                141m    10.104.14.54    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-etcd-2                           1/1     Running     1 (123m ago)     141m    10.104.13.56    4am-node16   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-datacoord-54577d75hx8l5   1/1     Running     0                141m    10.104.5.8      4am-node12   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-datanode-dd4ddfbdft4jk5   1/1     Running     1 (137m ago)     141m    10.104.4.147    4am-node11   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-indexcoord-78599c4p9dhp   1/1     Running     0                141m    10.104.14.44    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-indexnode-54b46fb857q6d   1/1     Running     0                141m    10.104.6.193    4am-node13   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-proxy-78d54b66b4-4h8qw    1/1     Running     2 (133m ago)     141m    10.104.6.192    4am-node13   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-querycoord-5f77dc4kx9w7   1/1     Running     2 (133m ago)     141m    10.104.14.45    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-querynode-5ddd8578xkpjz   1/1     Running     8 (72m ago)      141m    10.104.5.7      4am-node12   <none>           <none>
fouramf-stable-test-1691618400-1-milvus-rootcoord-6dbf54f7jvwnh   1/1     Running     2 (133m ago)     141m    10.104.5.6      4am-node12   <none>           <none>
fouramf-stable-test-1691618400-1-minio-0                          1/1     Running     0                141m    10.104.13.55    4am-node16   <none>           <none>
fouramf-stable-test-1691618400-1-minio-1                          1/1     Running     0                141m    10.104.1.216    4am-node10   <none>           <none>
fouramf-stable-test-1691618400-1-minio-2                          1/1     Running     0                141m    10.104.14.56    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-minio-3                          1/1     Running     0                141m    10.104.5.39     4am-node12   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-bookie-0                  1/1     Running     0                141m    10.104.16.190   4am-node21   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-bookie-1                  1/1     Running     0                141m    10.104.5.37     4am-node12   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-bookie-2                  1/1     Running     0                141m    10.104.1.221    4am-node10   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-bookie-init-gj67x         0/1     Completed   0                141m    10.104.14.42    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-broker-0                  1/1     Running     0                141m    10.104.17.124   4am-node23   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-proxy-0                   1/1     Running     0                141m    10.104.14.41    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-pulsar-init-54jvh         0/1     Completed   0                141m    10.104.14.43    4am-node18   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-recovery-0                1/1     Running     0                141m    10.104.16.186   4am-node21   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-zookeeper-0               1/1     Running     0                141m    10.104.4.154    4am-node11   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-zookeeper-1               1/1     Running     0                138m    10.104.1.235    4am-node10   <none>           <none>
fouramf-stable-test-1691618400-1-pulsar-zookeeper-2               1/1     Running     0                136m    10.104.5.56     4am-node12   <none>           <none>
```

client pod : fouramf-stable-test-1691618400-1387329173

**Comment by smellthemoon** [comment]
<img width="795" alt="1" src="https://github.com/milvus-io/milvus/assets/64083300/d1bc995a-d45b-4b21-8e6d-06cd86d19aad">
threads lost control.

**Comment by smellthemoon** [comment]
related with https://github.com/milvus-io/milvus/pull/26293

**Comment by smellthemoon** [comment]
@elstic pr merged, plz verify.

**Comment by elstic** [comment]
This issue has been fixed , verify image: master-20230813-09da953a
